sdk/core/azure-json/src/main/java/com/azure/json/JsonWriter.java

```diff
@@ -82,6 +107,23 @@ public abstract class JsonWriter implements Closeable {
      */
     public abstract JsonWriter writeFieldName(String fieldName);
 
+    /**
+     * Writes a {@link JsonCapable} object.
+     * <p>
+     * A value is always written no matter whether {@code value} is null, if null shouldn't be written this API call
+     * must be null guarded.
+     * <p>
+     * This API is used instead of {@link #writeJsonCapableField(String, JsonCapable)} when the value needs to be
+     * written to the root of the JSON value, as an element in an array, or after a call to
+     * {@link #writeFieldName(String)}.
+     *
+     * @param value {@link JsonCapable} object to write.
+     * @return The updated JsonWriter object.
+     */
+    public final JsonWriter writeJsonCapable(JsonCapable<?> value) {
```

Comment:
Perhaps it is because I'm not deeply embedded in this effort, but `writeJsonCapable` leads to the question 'what is `JsonCapable`?' Do you feel this is the right terminology to use throughout our API?